app/models/body_treatment.rb

```diff
@@ -0,0 +1,70 @@
+class BodyTreatment < ApplicationRecord
+  has_many :durations, dependent: :destroy
+  has_many :locations, dependent: :destroy
+  has_many :studios, through: :locations
+
+  validates :name, presence: true
+  validates :category, presence: true
+  validates :description, presence: true
+
+  # rubocop:disable Metrics/BlockLength
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Layout/LineLength
+  def self.body_treatments
+    treatments_query = BodyTreatment.select(:name, :category, :description, :city, :address, :duration, :price)
+      .joins(:studios, :durations).all.order(:name, :city, :address, :duration)
```

Comment:
This is the only part I meant to be scoped to the model

Reply:
Well, I cleaned the actions out.